### PR TITLE
simplify Docker setup

### PR DESCRIPTION
**Pull request by IonBazan**
This PR reduces number of Dockerfiles in the repo. It also adds `.gitattributes` file to ignore some files in dist. (see #1)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
-FROM php:7.1.3
```
Comment (thunderer) [inline docker/Dockerfile]: please move this file to `docker/php/Dockerfile`

File: docker-compose.yaml [hunk 2]
```diff
@@ -9,7 +9,10 @@ services:
     working_dir: /app
 
   php:
-    build: docker/php${PHP_VERSION}-xdebug
+    build:
+      context: docker
```
Comment (thunderer) [inline docker-compose.yaml]: please change this to `docker/php`

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/tests export-ignore
+/docker export-ignore
+/.gitattributes export-ignore
+/.gitignore export-ignore
+/.travis.yml export-ignore
+/docker-compose.yaml export-ignore
+/infection.json export-ignore
+/Makefile export-ignore
+/phpunit.xml.dist export-ignore
+/psalm.xml export-ignore
```
Comment (thunderer) [inline .gitattributes:10]: please sort these entries by name, ie. move `tests` and `docker` to their places

**Review by thunderer — CHANGES_REQUESTED** [review]

**Comment by thunderer** [comment]
Thank you very much for this PR, good job! I'll look into that Psalm error shortly.

**Comment by thunderer** [comment]
@IonBazan I looked at the Psalm issue, please disable this case by adding a `/** @psalm-suppress RedundantCondition */` docblock above line 258 in EnumTrait.

**Comment by IonBazan** [comment]
Thanks for the tips, fixed!

BTW, I see this package has very nice mutation score (88%). Maybe it is worth adding a badge for that: https://dashboard.stryker-mutator.io/

**Comment by thunderer** [comment]
@IonBazan good idea, I'll take a look at it. Meanwhile, I need to finally push my local branch which makes it even better. :)

**Review by thunderer — APPROVED** [review]

**Pull request merged**